### ISSUE TITLE
False positive no unused return values for Enum functions

### ISSUE THREAD

**Issue by ourway** [rrrene/credo#694]
### Precheck

* For proposing a new feature, be sure to name your issue "Proposal: ..."
* For bugs, please do a quick search and make sure the bug has not yet been reported here

### Environment

* Credo version (`mix credo -v`): 1.1.2
* Erlang/Elixir version (`elixir -v`): 1.9.1
* Operating system: 4.19.66-1-MANJARO

### What were you trying to do?
```elixir
  @impl true
  def handle_info({:resume_crawl}, state) do
    state.links_map |> Enum.map(fn section -> send(self(), {:resume_task_crawl, section.name}) end)

    {:noreply, state}
  end

```

### Expected outcome

No warnings

### Actual outcome

```
Warnings - please take a look                                                 

There should be no unused return values for Enum functions.  
```

**Comment by NobbZ** [comment]
In general, `Enum.map/2` is used to transform a list, its result is **ment** to be used.

If you do process a list item by item solely for its side effects, then `Enum.each/2` is what you want.
